### PR TITLE
#2968 Missing R Host produces error that is not informative

### PR DESCRIPTION
**Pull request by MikhailArkhipov**

### CODE REVIEW

File: src/Host/Client/Test/Session/RSessionTest.cs [hunk 1]
```diff
@@ -183,7 +183,7 @@ public partial class RSessionTest : IDisposable {
             var session = new RSession(0, _testMethod.Name, brokerClient, new AsyncReaderWriterLock().CreateExclusiveReaderLock(), () => { });
             Func<Task> start = () => session.StartHostAsync(new RHostStartupInfo(), null, 10000);
 
-            start.ShouldThrow<RHostBrokerBinaryMissingException>();
+            start.ShouldThrow<ComponentBinaryMissingException>();
```
Comment (AlexanderSher) [inline src/Host/Client/Test/Session/RSessionTest.cs]: Should it be `RHostBinaryMissingException` in this case?

Comment (MikhailArkhipov) [inline src/Host/Client/Test/Session/RSessionTest.cs]: Yes

File: src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs [hunk 2]
```diff
@@ -283,9 +283,11 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
             // From R docs:  Valid values are 10...10000 with default normally 80.
             _terminalWidth = Math.Max(10, Math.Min(10000, width));
 
-            Session.OptionsSetWidthAsync(_terminalWidth)
-                .SilenceException<RException>()
-                .DoNotWait();
+            if (Session.IsHostRunning) {
+                Session.OptionsSetWidthAsync(_terminalWidth)
```
Comment (AlexanderSher) [inline src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs]: `Session.IsHostRunning` may change right after you've checked it. We should add `SilenceException<RHostDisconnectedException>()` to the `OptionsSetWidthAsync` call instead.

Comment (MikhailArkhipov) [inline src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs]: That think is extension to `IRExpressionEvaluator` so it doesn't have that information. I guess I can just leave this alone

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**